### PR TITLE
[codeintel] RFC 619: refactor of uploads background jobs 

### PR DESCRIPTION
**Pull request by Numbers88s**
Refactor of uploads background jobs as part of our ongoing efforts around RFC 619.

## Test plan
Ensure previous test continue to work.

### CODE REVIEW

File: internal/codeintel/uploads/internal/store/store_commits_test.go [hunk 1]
```diff
@@ -164,8 +165,40 @@ func TestDeleteSourcedCommits(t *testing.T) {
 	}
 }
 
+func TestMarkRepositoryAsDirty(t *testing.T) {
```
Comment (efritz) [inline internal/codeintel/uploads/internal/store/store_commits_test.go:168]: You renamed Mark -> Set. Can we be consistent on the test names here?

File: internal/codeintel/uploads/init.go [hunk 2]
```diff
@@ -19,24 +24,40 @@ var (
...
+		oc := func(name string) *observation.Context {
+			return &observation.Context{
+				Logger:     lg(name),
+				Tracer:     &trace.Tracer{Tracer: opentracing.GlobalTracer()},
+				Registerer: prometheus.DefaultRegisterer,
+			}
 		}
-		svc = newService(store, observationContext)
+
+		codeIntelDB := mustInitializeCodeIntelDB(lg("lsifstore"))
```
Comment (efritz) [inline internal/codeintel/uploads/init.go:43]: @Numbers88s OK I think this is what I was worried about (the dbstore vs lsifstore handle), but it seems to be correct.

**Review by efritz — APPROVED** [review]
Approving to unblock - looks mostly mechanical. Could you have @Strum355 run through this locally to ensure that the dashboards are still set up correctly?

File: internal/codeintel/uploads/background/cleanup/unknown_commits.go [hunk 3]
```diff
@@ -13,9 +13,6 @@ import (
 )
 
 func (j *janitor) HandleUnknownCommit(ctx context.Context) (err error) {
-	tx, err := j.dbStore.Transact(ctx)
-	defer func() { err = tx.Done(err) }()
```
Comment (Strum355) [inline internal/codeintel/uploads/background/cleanup/unknown_commits.go:17]: @efritz does removing the transaction here cause correctness issues?

File: internal/codeintel/uploads/internal/lsifstore/lsifstore.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import (
+	"context"
+
+	"github.com/sourcegraph/sourcegraph/internal/codeintel/stores"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+)
+
+type LsifStore interface {
+	Clear(ctx context.Context, bundleIDs ...int) (err error)
```
Comment (Strum355) [inline internal/codeintel/uploads/internal/lsifstore/lsifstore.go:12]: I would probably see if theres a better name for this, or else give it a doc comment. Without looking further, Im not immediately sure what it does (clears LSIF data for the given uploads probably?)

File: internal/codeintel/uploads/internal/store/scan.go [hunk 5]
```diff
@@ -13,3 +18,150 @@ func scanUpload(s dbutil.Scanner) (upload shared.Upload, err error) {
...
+		var count int
+		if err := rows.Scan(&id, &count); err != nil {
+			return nil, err
+		}
+
+		visibilities[id] = count
+	}
+
+	return visibilities, nil
+}
```
Comment (Strum355) [inline internal/codeintel/uploads/internal/store/scan.go:80]: I think most of the functions here could be replaced by the constructs in `internal/database/basestore/rows.go` to clean things up a bit :smile:

**Review by Strum355 — COMMENTED** [review]
Gave it a first pass through, need to go over the rest soon:tm:

File: internal/codeintel/uploads/background/cleanup/unknown_commits.go (hunk 3 above)
Comment (efritz) [reply internal/codeintel/uploads/background/cleanup/unknown_commits.go:17]: It shouldn't - the previous txn wrapped the read + updates to help with concurrent updates from multiple workers, so I think at worst we'll ask gitserver about the same repo twice.

Might be worth it to check the UpdateSourcedCommits method to re-check the conditions where it updates, but that might even be overkill.

**Pull request merged**